Loop/Managers/DeviceDataManager.swift

```diff
@@ -1357,84 +1366,92 @@ extension Notification.Name {
 }
 
 // MARK: - Remote Notification Handling
-extension DeviceDataManager {
+
+extension DeviceDataManager: RemoteCommandHandler {
     
     func handleRemoteNotification(_ notification: [String: AnyObject]) {
         Task {
-            let backgroundTask = await beginBackgroundTask(name: "Remote Data Upload")
-            await handleRemoteNotification(notification)
+            log.default("Remote Notification: Handling notification %{public}@", notification)
+            
+            guard FeatureFlags.remoteCommandsEnabled else {
+                log.error("Remote Notification: Remote Commands not enabled.")
+                return
+            }
+            
+            let backgroundTask = await beginBackgroundTask(name: "Handle Remote Notification")
+            do {
+                try await remoteDataServicesManager.handleRemoteNotification(notification)
+            } catch {
+                log.error("Remote Notification: %{public}@. Error: %{public}@", String(describing: notification), String(describing: error))
+            }
+            
             await endBackgroundTask(backgroundTask)
+            log.default("Remote Notification: Finished handling")
         }
     }
     
```

Comment:
REMOTE 2.0: WILL REMOVE PRE-MERGE

This will ask the plugins to check for new remote commands and handle them.